server.js

```diff
@@ -1111,44 +1112,79 @@ function objectWebServer() {
     });
 
     webServer.use('/obj', function (req, res, next) {
-
+        
+       
+     
         var urlArray = req.originalUrl.split('/');
         urlArray.splice(0, 1);
         urlArray.splice(0, 1);
         if (urlArray[1] === 'frames') {
+            var objectKey = utilities.readObject(objectLookup, urlArray[0]);
+            var frameKey = utilities.readObject(objectLookup, urlArray[0]) + urlArray[2];
+            var thisFrame = getFrame(objectKey, frameKey);
+            
+            var toolpath = null;
+            
+            if(thisFrame !== null){
+                if(thisFrame.hasOwnProperty('tool')){
+                    if(thisFrame.tool.hasOwnProperty('addon') && thisFrame.tool.hasOwnProperty('interface') && thisFrame.tool.hasOwnProperty('tool')){
+                        toolpath =  __dirname+'/addons/'+thisFrame.tool.addon+'/interfaces/'+thisFrame.tool.interface+'/tools/'+thisFrame.tool.tool;
+                    }
+                }
+            }
+
             urlArray.splice(1, 1);
         }
+        
+        var switchToInteraceTool = true;
+        if(!toolpath) switchToInteraceTool = false;
 
         if ((urlArray[urlArray.length - 1] === 'target.dat' || urlArray[urlArray.length - 1] === 'target.jpg' || urlArray[urlArray.length - 1] === 'target.xml')
             && urlArray[urlArray.length - 2] === 'target') {
             urlArray[urlArray.length - 2] = identityFolderName + '/target';
+            switchToInteraceTool = false;
         }
 
         if ((urlArray[urlArray.length - 1] === 'memory.jpg' || urlArray[urlArray.length - 1] === 'memoryThumbnail.jpg')
             && urlArray[urlArray.length - 2] === 'memory') {
             urlArray[urlArray.length - 2] = identityFolderName + '/memory';
+            switchToInteraceTool = false;
         }
 
         if ((urlArray[urlArray.length - 2] === 'videos') && urlArray[urlArray.length - 1].split('.').pop() === 'mp4') {
             urlArray[urlArray.length - 2] = identityFolderName + '/videos';
+            switchToInteraceTool = false;
         }
 
         var newUrl = '';
+        var newToolUrl = '';
         for (let i = 0; i < urlArray.length; i++) {
             newUrl += '/' + urlArray[i];
+            
         }
 
+        if(toolpath !== null) {
+            for (let i = 2; i < urlArray.length; i++) {
```

Comment:
It would be nice to get comments explaining why numbers like this 2 and the above -1 -2 -3 come from since if there is a future attempt to touch this code or refactor it to be more streamlined every single magic number will require thought and reverse-engineering.